bridge/client/app/_views/ktb-settings-view/ktb-integration-view/ktb-modify-uniform-subscription/ktb-modify-uniform-subscription.component.ts

```diff
@@ -348,6 +357,39 @@ export class KtbModifyUniformSubscriptionComponent implements OnDestroy {
     this._changeDetectorRef.detectChanges();
   }
 
+  // @HostListener allows us to also guard against browser refresh, close, etc.
+  @HostListener('window:beforeunload', ['$event'])
+  public canDeactivate(_$event?: BeforeUnloadEvent): Observable<boolean> {
+    if (!this.overrideChangeGuard && (this.subscriptionForm.touched || this.webhookFormTouched)) {
+      this.showNotification();
+      return this.pendingChangesSubject.asObservable();
+    }
+    return of(true);
+  }
+
+  public reject(): void {
+    this.pendingChangesSubject.next(false);
+    this.hideNotification();
+  }
+
+  public reset(): void {
+    this.pendingChangesSubject.next(true);
+    this.hideNotification();
+  }
+
+  public showNotification(): void {
+    this.unsavedDialogState = 'unsaved';
+
+    document.querySelector('div[aria-label="Dialog for notifying about unsaved data"]')?.classList.add('shake');
+    setTimeout(() => {
+      document.querySelector('div[aria-label="Dialog for notifying about unsaved data"]')?.classList.remove('shake');
+    }, 500);
```

Comment:
Set the class more reactive:
```ts
this.shakeIt = true;
setTimeout(() => this.shakeIt = false, 500);
```
```html
<dt-confirmation-dialog [class.shake]="shakeIt"
```

Reply:
This is not possible as Barista creates a new element inside its `overlay` div.